pkg/kubelet/lifecycle/handlers.go

```diff
@@ -92,7 +93,7 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 		return portReference.IntValue(), nil
 	}
 	portName := portReference.StrVal
-	port, err := strconv.Atoi(portName)
+	port, err := utilsnet.ParsePort(portName, false)
```

Comment:
just curious, is 0 allowed here?

Reply:
By reviewing this change again, I prefer to cut it out from this PR. The port is here is for containers instead of hosts.